### PR TITLE
Improve coverage of mypy checks, and add mypy to CI

### PR DESCRIPTION
**Pull request by jacksonj04**
At the moment mypy under `fab test` is only checking the `ds_caselaw_editor_ui` folder, but we also implement a lot of stuff in the `judgments` folder and this isn't being analysed.

In addition, mypy isn't being run at all in CI. Add a new task to the workflow to run it.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -35,6 +35,24 @@ jobs:
...
+      - name: Copy .env.example file
+        uses: canastro/copy-file-action@master
+        with:
+          source: ".env.example"
+          target: ".env"
+
+      - name: Build the Stack
+        run: docker-compose build postgres django
+
+      - name: Run mypy
```
Comment (anthonyhashemi) [inline .github/workflows/ci.yml:53]: do we need to build the stack to run mypy?

Comment (jacksonj04) [reply .github/workflows/ci.yml:53]: Annoyingly, yes. I'm debating moving it to just a step in the regular tests so we're not building it twice.

**Comment by dragon-dxw** [comment]
Was going to suggest adding mypy to pre-commit, but it's a pain. Got as far as

 ```yaml
  - repo: https://github.com/pre-commit/mirrors-mypy
    rev: v1.0.1
    hooks:
      - id: mypy
        additional_dependencies:
          - django-stubs
          - django
          - djangorestframework
          - django-environ
      ```
      
      before giving up because it's running the code and complaining about environment variables.

**Comment by jacksonj04** [comment]
@dragon-dxw that's _exactly_ where I got to before giving up as well!

File: judgments/utils/__init__.py [hunk 2]
```diff
@@ -134,10 +134,8 @@ def render_versions(decoded_versions):
 
 
 def extract_version(version_string: str) -> int:
-    try:
-        return int(re.search(VERSION_REGEX, version_string).group(1))
-    except AttributeError:
-        return 0
+    result = re.search(VERSION_REGEX, version_string)
```
Comment (dragon-dxw) [inline judgments/utils/__init__.py:137]: I have no idea what this code was meant to be doing.

**Review by dragon-dxw — APPROVED** [review]
I'm wary that if Mypy is only running in CI, it's going to be a pain to fix Mypy problems. Would very much like a pre-commit hook.

I hope it is worth the extra pain :)

File: judgments/utils/__init__.py (hunk 2 above)
Comment (jacksonj04) [reply judgments/utils/__init__.py:137]: it looks like it was supposed to handle the error case of "there are no matches" by returning `0`.

**Pull request merged**